### PR TITLE
nixos/tests/nextcloud: Fix tests, fix broken webdav url

### PR DESCRIPTION
**Pull request by onny**
###### Description of changes

Fixes tests for me by changing webdav url.

Starting with Rclone v1.63, which is used in the Nextcloud tests for
synchronization, the client relies on the correct WebDAV endpoint url,
see https://github.com/rclone/rclone/issues/7103

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by Ma27** [comment]
not sure if I agree with all of the changes, so please wait for a bit until I have time for a review.

**Comment by winterqt** [comment]
Yeah, at the very least an explanation of why you made these seemingly unrelated changes would be nice... are you sure all of these are required? What was the issue in the first place?

**Comment by onny** [comment]
Hi @winterqt the Nextcloud tests were broken for me, with error messages of rclone that the webdav endpoint changed. These changes fixes the tests for me :)

File: nixos/tests/nextcloud/with-declarative-redis-and-secrets.nix [hunk 1]
```diff
@@ -1,6 +1,6 @@
 args@{ nextcloudVersion ? 27, ... }:
 (import ../make-test-python.nix ({ pkgs, ...}: let
-  username = "custom_admin_username";
+  adminuser = "root";
```
Comment (Ma27) [inline nixos/tests/nextcloud/with-declarative-redis-and-secrets.nix:3]: Was this perhaps part of your debugging that you accidentally pushed? Apparently the issue was that we need a different URL now for rclone.
Also, I think that this is somewhat useful, i.e. to have the case tested with a non-standard admin user.

File: nixos/tests/nextcloud/basic.nix [hunk 2]
```diff
@@ -14,12 +14,12 @@ in {
     client = { ... }: {
       services.davfs2.enable = true;
       system.activationScripts.davfs2-secrets = ''
-        echo "http://nextcloud/remote.php/webdav/ ${adminuser} ${adminpass}" > /tmp/davfs2-secrets
+        echo "http://nextcloud/remote.php/dav/files/${adminuser} ${adminuser} ${adminpass}" > /tmp/davfs2-secrets
```
Comment (Ma27) [inline nixos/tests/nextcloud/basic.nix:17]: I understand that this is apparently needed to fix the test. But please leave an explanation in the PR's op and the commit message on the details (i.e. what in rclone has changed that this became necessary)?

**Review by Ma27 — COMMENTED** [review]
If I had to guess this is related to the rclone bump in 560dae862f5c903df830e24a3fbef0c7d2008cf5 (the "First broken build" in https://hydra.nixos.org/build/229621017 seems to confirm that), my comments assume that this is the case, but I haven't checked closely.

I'm also wondering if we want to add the nextcloud/basic test to rclone's `passthru.test` because this test not only tests nextcloud as server, but also relies pretty much on rclone. That way we'd spot those issues earlier and we effectively have a regression test for rclone for free :)

File: nixos/tests/nextcloud/basic.nix (hunk 2 above)
Comment (SuperSandro2000) [reply nixos/tests/nextcloud/basic.nix:17]: I think other people described this issue in https://github.com/rclone/rclone/issues/7103

Comment (Ma27) [reply nixos/tests/nextcloud/basic.nix:17]: Yeah, a tldr (i.e. one short sentence) and a link to the upstream issue is perfectly sufficient IMHO.

Comment (EloraDarling) [reply nixos/tests/nextcloud/basic.nix:17]: I'm not sure but I wanted to chime in that there seems to be some issues with Nextcloud and the current released rclone beta. Hopefully this PR is aiming to fix that. I'll move files on the Nextcloud backend, and running rclone lsd doesn't seem to reflect those changes to where the files are located, yet the backend confirms they have been moved.

**Comment by onny** [comment]
Thanks @Ma27 added a more verbose commit message

**Review by Ma27 — APPROVED** [review]

**Comment by onny** [comment]
Great thank yoi :)

**Pull request merged**